### DISCUSSION TITLE
Controller read resource another controller

### DISCUSSION THREAD

**Discussion by Tomelin** [kubernetes-sigs/kubebuilder#3172] [Support] [answered]
I'm creating the controller with Kubebuilder.  The first controller  has resource and the second controller doesn't resource.

My Question, How can second controller read the resource from first controller?  or. How I get the namespace that the controller will be create? different the namespace that resource will create?

Thanks!

**Comment by camilamacedo86** [comment, marked as answer]
Yes, you can achieve that.
You're looking to create a controller primarily responsible for managing a principal Kind (e.g., MyKind). This will be your main controller for this Kind.

However, to implement this solution, you'll also need to manage a second Kind created in your project. You'll watch this, and manage it similarly to how you'd handle a deployment.

The code utilizes the outdated watch syntax. Now, it's recommended to use the builder pattern. While the current layout is no longer supported, you can find an example in the [postgresql-operator project](https://github.com/dev4devs-com/postgresql-operator). In this project, the Kind responsible for creating the cronjob observes the Kind that represents the database on the cluster.


Closing this one as sorted out.
If you need please feel free to re-open.
